### ISSUE TITLE
Ack prints extra newlines if pattern can match the newline and --color is on

### ISSUE THREAD

**Issue by petdance** [beyondgrep/ack1#83]
Reported by krawczyk, Sep 12, 2008

If the pattern supplied to ack can match the newline and highlighting is
turned on, extra newlines are printed.

I believe the proper fix is to make sure that newline is stripped out of
the string before the color is applied, somewhere around line 1187 of
App::Ack 1.86.

Reproduce: 

$ ack 'Lester[^Z]*$' `which ack`
Andy Lester, C<< <andy at petdance.com> >>

Copyright 2005-2008 Andy Lester, all rights reserved.

$

**Comment by funkopotamus-zz** [comment]
I can confirm this issue, and it is rather annoying.  Any time I want to match a pattern at the end of a line, I cannot avoid the extra newlines in the output.  I am trying to teach myself perl so that I can attempt your proposed fix.

**Comment by chocolateboy** [comment]
It looks like this (ack not chomping line endings before matching) is the issue I've encountered with negative patterns e.g. `\W`, `\D` and `[^a-zA-Z0-9]`. All of these match the trailing line-ending character(s), rendering these patterns effectively useless.

Another unexpected consequence is that `ack '\s'` matches all lines.

I've posted [more details](https://groups.google.com/forum/#!topic/ack-users/8DIrUP02vG0) on the ack-users forum. As mentioned there, `grep -P` does not share this behaviour.
